src/transfers/monitor.hpp

```diff
@@ -208,6 +212,8 @@ class Monitor {
     Timestamp start;
     size_t expected;
     size_t transferred;
+
+    bool path_known = false;
```

Comment:
Is it worth the effort, or are we OK conflating empty path and null path here?

I guess `optional<char [FILE_PATH_BUFFER_LEN]>` would be too much of a mental exercise? 😈

Reply:
I agree we could just omit the bool, but AFAIK you can't put an array type into a optional. (taken from [cppreference](https://en.cppreference.com/w/cpp/utility/optional))
![image](https://user-images.githubusercontent.com/33360155/210220249-42c9f8eb-c119-4c84-b46f-558f9bb1c5e5.png)